specification/cosmos-db/resource-manager/Microsoft.DocumentDB/stable/2021-03-15/examples/CosmosDBDatabaseAccountCreateMax.json

```diff
@@ -0,0 +1,205 @@
+{
+  "parameters": {
+    "accountName": "ddb1",
+    "resourceGroupName": "rg1",
+    "api-version": "2021-03-15",
+    "subscriptionId": "subid",
+    "createUpdateParameters": {
+      "location": "westus",
+      "tags": {},
+      "kind": "MongoDB",
+      "identity": {
+        "type": "SystemAssigned,UserAssigned",
+        "userAssignedIdentities": {
+          "/subscriptions/fa5fc227-a624-475e-b696-cdd604c735bc/resourceGroups/eu2cgroup/providers/Microsoft.ManagedIdentity/userAssignedIdentities/id1": {}
+        }
+      },
+      "properties": {
+        "databaseAccountOfferType": "Standard",
+        "ipRules": [
+          {
+            "ipAddressOrRange": "23.43.230.120"
+          },
+          {
+            "ipAddressOrRange": "110.12.240.0/12"
+          }
+        ],
+        "isVirtualNetworkFilterEnabled": true,
+        "virtualNetworkRules": [
+          {
+            "id": "/subscriptions/subId/resourceGroups/rg/providers/Microsoft.Network/virtualNetworks/vnet1/subnets/subnet1",
+            "ignoreMissingVNetServiceEndpoint": false
+          }
+        ],
+        "publicNetworkAccess": "Enabled",
+        "locations": [
+          {
+            "failoverPriority": 0,
+            "locationName": "southcentralus",
+            "isZoneRedundant": false
+          },
+          {
+            "failoverPriority": 1,
+            "locationName": "eastus",
+            "isZoneRedundant": false
+          }
+        ],
+        "consistencyPolicy": {
+          "defaultConsistencyLevel": "BoundedStaleness",
+          "maxIntervalInSeconds": 10,
+          "maxStalenessPrefix": 200
+        },
+        "keyVaultKeyUri": "https://myKeyVault.vault.azure.net",
+        "defaultIdentity": "FirstPartyIdentity",
```

Comment:
you support getting the KV key with your RPs first party identity?  How do you secure that so users can't point to a KV they don't have any access to which another customer may have given your first party app access to?

Reply:
Hi, we always do "key vault access validation" to check if the new default identity to be changed into will have access to key vault or not. If NO, then a bad request will be thrown directly to the customer. 

When the db account is created, it's by default to use "FIrstPartyIdentity", which is an globally shared Service Principal, then later the users can choose to switch to a db account level identity like "SystemAssignedIdentity". Once the switch is done, then the db account will not count on the "FirstPartiyIdentity" any longer.

Reply:
Please make sure this approach goes through some level of security reviews / threat modeling